plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/ProgramConverter.kt

```diff
@@ -98,9 +101,12 @@ class ProgramConverter(val session: FirSession, override val config: PluginConfi
 
         // Note: it is important that `body` is only set after `embedUserFunction` is complete, as we need to
         // place the embedding in the map before processing the body.
-        embedUserFunction(declaration.symbol, signature).apply {
+        val userFunc = embedUserFunction(declaration.symbol, signature).apply {
             body = stmtCtx.convertMethodWithBody(declaration, signature, returnTarget)
         }
+
+        // For now this will just traverse the ExpEmbedding Tree and check whether the asserts are true
```

Comment:
At this point you cannot check whether the asserts are true, since it's up to Viper to do that.  What we discussed was to check whether the assert conditions are pure.